web/meetup-facilitator/src/components/Recommendation.jsx

```diff
@@ -0,0 +1,217 @@
+        month: 'long',
+        day: 'numeric',
+        year: 'numeric',
+        hour: 'numeric',
+        minute: 'numeric',
+        hour12: true,
+      };
+      return dateObject.toLocaleString('en-US', options);
+    });
+
+    setPopupTimes(timesArray);
+  };
+
+  const closeModal = () => {
+    setIsModalOpen(false);
+  };
+
+  //error check if no recommendation
+  if (recommendations.length === 0) {
+    return (
+      <div className="flex items-center justify-center h-screen">
+        <p className="text-xl text-gray-500">
+          We couldn't find a place for your group to meet. Try changing your activities, times, or location range!
+        </p>
+      </div>
+    );
+  }
+
+  //webpage format
+  return (
+    <div className="flex">
+      <div className="flex-row w-1/2 px-10"> 
+        <h2 className="py-8 text-2xl">Recommendation Page</h2>
+
+        {recommendations.map((recommendation, index) => (
+          <div
+            key={index}
+            className={`py-2 mb-6 h-44 rounded-lg bg-slate-500 cursor-pointer ${
+              selectedDiv === index ? 'border-2 border-green-500' : ''
+            }`}
+            onClick={() => openModal(index)}
+          >
+
+            <div className="flex flex-col items-center h-full">
+              <button
+                onClick={() => window.open(recommendation.place_url, '_blank')}
+                className="text-white text-xl bg-slate-500 mb-4" 
+              >
+                {recommendation.place_name}
+              </button>
+              <div className="flex flex-col items-center">
+                <div className="mb-2">
+                  <strong>Type of Activity:</strong> <br />{recommendation.activity_id}
+                </div>
+                <div className=" bg-green-700 w-8 h-8 rounded-full flex items-center justify-center">
+                  {votesData.filter(vote => vote.rec_id === recommendation.id).length}
+                </div>
+              </div>
+            </div>
+          </div>
+        ))}
+      </div>
+            
+      <div className="w-2/3">
+        <RecGoogleMap coordinates={sampleCoordinates} recommendations={recommendations} />
+      </div>
+
+      {isModalOpen && (<RecPopup
+      closeModal={closeModal} 
+      times={popupTimes}
+      onSelect={onSelectHandler}
+      />
+      )}
+        <button
+        className={`fixed bottom-4 right-4 bg-green-800 text-white py-2 px-4 rounded-md cursor-pointer
+        }`}
+        onClick={() => {
+
+          if (selectedDiv !== null && selectedDiv !== undefined && selectedTime !== null && selectedTime !== undefined) {
+            // The logic to save the vote
+            const selectedRecommendation = recommendations[selectedDiv];
+            const voteData = {
+              rec_id: {
+                recommendation_id: selectedRecommendation.id,
+                selected_time: selectedTime,
+              },
+              user_id: userdata,
+              group_id: group_id,
```

Comment:
`rec_id` is just the id of the recommendation, so `selectedRecommendation.id`, and `user_id` should be `userdata.user_id`. We don't need to pass the `group_id` since it's given in the URL. I had forgotten about the time in the Votes model, thanks for noticing, we will need to make a new field for it in the API model. Overall I think we only need `rec_id`, `selected_time`, and `user_id`.